### PR TITLE
Feature/invites

### PR DESCRIPTION
**Pull request by ganfra**
Start handling invites. It will require to adjust UI, waiting for Nad inputs.

### CODE REVIEW

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/database/helper/ChunkEntityHelper.kt [hunk 1]
```diff
@@ -86,10 +86,9 @@ internal fun ChunkEntity.add(roomId: String,
                              isUnlinked: Boolean = false) {
 
     assertIsManaged()
-    if (event.eventId.isNullOrEmpty() || this.events.fastContains(event.eventId)) {
+    if (event.eventId != null && events.fastContains(event.eventId)) {
```
Comment (BillCarsonFr) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/database/helper/ChunkEntityHelper.kt]: It's not doing the same thing (it's a fix?)

File: vector/src/main/res/values/text_appearances.xml [hunk 2]
```diff
@@ -1,5 +1,40 @@
...
+  ~ limitations under the License.
+  -->
 
+<resources xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <style name="TextAppearance.Vector.Title" parent="TextAppearance.AppCompat">
+        <item name="android:textSize">16sp</item>
+        <item name="android:fontFamily">sans-serif-medium</item>
+        <item name="android:textStyle">normal</item>
+        <item name="android:textColor">@color/black_87</item>
```
Comment (BillCarsonFr) [inline vector/src/main/res/values/text_appearances.xml]: not dark theme friendly

Comment (ganfra) [reply vector/src/main/res/values/text_appearances.xml]: Yep, I'll do it later, we need a total theme rework !

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/database/helper/ChunkEntityHelper.kt (hunk 1 above)
Comment (ganfra) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/database/helper/ChunkEntityHelper.kt]: Yep, it's a fix for handling empty event id (it happens inside invitation room timeline)

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/RoomAPI.kt [hunk 3]
```diff
@@ -156,4 +161,24 @@ internal interface RoomAPI {
...
+
+    /**
+     * Join the given room.
+     *
+     * @param roomId  the room id
+     * @param params the request body
+     */
+    @POST(NetworkConstants.URI_API_PREFIX_PATH_R0 + "rooms/{roomId}/join")
+    fun join(@Path("roomId") roomId: String,
+             @Body params: Map<String, String>): Call<Unit>
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/RoomAPI.kt]: Is it possible to create dedicated data class for `params`? Current usage is empty HashMap...

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/RoomAPI.kt [hunk 4]
```diff
@@ -156,4 +161,24 @@ internal interface RoomAPI {
...
+
+    /**
+     * Leave the given room.
+     *
+     * @param roomId  the room id
+     * @param params the request body
+     */
+    @POST(NetworkConstants.URI_API_PREFIX_PATH_R0 + "rooms/{roomId}/leave")
+    fun leave(@Path("roomId") roomId: String,
+              @Body params: Map<String, String>): Call<Unit>
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/RoomAPI.kt]: Is it possible to create dedicated data class for `params`? Current usage is empty HashMap...

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/RoomAPI.kt (hunk 3 above)
Comment (ganfra) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/RoomAPI.kt]: I will do that later, at the moment it was directly taken from riot legacy

**Pull request merged**